pyam/core.py

```diff
@@ -405,6 +405,28 @@ def validate(self, criteria={}, exclude=False):
 
             return df
 
+    def conv_unit(self, unit_conv, inplace=False):
+        """ Converts units based on provided unit conversion factors
+
+        Parameters
+        ----------
+        unit_conv: dict
+            for each unit for which a conversion should be carried out, provide current unit and target unit and conversion factor
+            {<current unit>: [<target unit>, <conversion factor>]}
+        inplace: bool, default False
+            if True, do operation inplace and return None
+        """
+        ret = copy.deepcopy(self) if not inplace else self
+        for unt in unit_conv:
```

Comment:
suggest to rename `unt` to `unit` for readability

Reply:
suggest looping as 

```
for current_unit, (new_unit, factor) in unit_conv.items():
    do stuff
```